API_MANIFESTO.md

```diff
@@ -0,0 +1,125 @@
+# The API Manifesto
+
+# Infinispan API dependency module
+
+The ```infinispan-api``` dependency will contain the public API. This module will be independent from the other modules. 
+Dependency with implementations will be **logical**. This will keep the modularity needed for the different use cases.
```

Comment:
@karesti what do you mean with _logical dependency_?

Reply:
I mean that you need the implementation dependency in the classpath, but you might have all the interfaces available. This is to avoid having all the jars of the planet but avoid overcomplicating the API to avoid adding optional dependencies (search, caching and so on)